src/utils/auth.js

```diff
@@ -1,30 +1,36 @@
-const BASE_URL = "http://localhost:3000";
+const BASE_URL =
+  process.env.NODE_ENV !== "production"
+    ? "https://api.sam-news-explorer.students.nomoredomainssbs.ru"
+    : "https://localhost:3000";
 
-export const register = (email, password, username) => {
+export const register = (email, password, name) => {
   return fetch(`${BASE_URL}/signup`, {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
       "Access-Control-Allow-Origin": BASE_URL,
     },
-    body: JSON.stringify({ email, password, username }),
+    body: JSON.stringify({ email, password, name }),
   }).then((res) => {
-    if (res.status === 201) {
+    if (res.status === 201 || 409 || 400 || 429 || 500) {
       return res.json();
     }
-  });
+  }).catch((res) => { return res })
```

Comment:
It’s better not to use `catch` in `api files` because you need to `catch` possible errors at the very end of any request and this end is in  (`App.js`)  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.